src/audio/destination_node.rs

```diff
@@ -1,16 +1,49 @@
-use audio::node::{AudioNodeEngine, BlockInfo};
 use audio::block::Chunk;
+use audio::node::{AudioNodeEngine, BlockInfo};
+use audio::render_thread::AudioRenderThreadMsg;
+use audio::sink::AudioSink;
+use std::sync::mpsc::Sender;
 
-pub struct DestinationNode {}
+#[cfg(feature = "gst")]
+use backends::gstreamer::audio_sink::GStreamerAudioSink;
+
+pub struct DestinationNode {
+    sink: Box<AudioSink>,
+}
 
 impl DestinationNode {
-    pub fn new() -> Self {
-        Self {}
+    pub fn new() -> Result<Self, ()> {
+        #[cfg(feature = "gst")]
+        let sink = GStreamerAudioSink::new()?;
+
+        Ok(Self {
+            sink: Box::new(sink),
+        })
+    }
+
+    pub fn init(&self, sample_rate: f32, sender: Sender<AudioRenderThreadMsg>) -> Result<(), ()> {
+        self.sink.init(sample_rate, sender)?;
+        Ok(self.sink.play())
+    }
+
+    pub fn has_enough_data(&self) -> bool {
+        self.sink.has_enough_data()
     }
 }
 
 impl AudioNodeEngine for DestinationNode {
-    fn process(&mut self, inputs: Chunk, _: &BlockInfo) -> Chunk {
-        inputs
+    fn process(&mut self, inputs: Chunk, _: &BlockInfo) -> Option<Chunk> {
```

Comment:
I don't think we need to return Option here, Chunk::default is empty.

Also, can we switch back to the old sink model? My graph PR needs it to work since there's some extra work that may need to be done in between.

Reply:
Ok. Since the PR was approved and you didn't comment any further, I assumed you were ok with merging this. I'll revert it.

Reply:
6131d39e6bdc9e300231c768a0b389e097c68fd9